src/display_buffer.cc

```diff
@@ -288,6 +288,11 @@ void DisplayBuffer::optimize()
 
 DisplayLine parse_display_line(StringView line)
 {
+    return parse_display_line_with_atoms(line, HashMap<String, DisplayLine>{});
+}
+
+DisplayLine parse_display_line_with_atoms(StringView line, HashMap<String, DisplayLine> atoms)
```

Comment:
I dont think we need the two functions, just `parse_display_line` taking a hash map argument with a defaultu value of `= {}`.